### PR TITLE
Database

### PR DESCRIPTION
**Pull request by riittagerhards**
Setup connection to MongoDB, test the connection with post and get commands in index.ts  and samples.http

image

part of #9

### CODE REVIEW

**Review by lajanzen — APPROVED** [review]
LGTM!

File: samples.http [hunk 1]
```diff
@@ -0,0 +1,13 @@
+### GET all entries
+GET http://localhost:3000/api/entries HTTP/1.1
```
Comment (lmachens) [inline samples.http:2]: The default server port is `3001`:

```suggestion
GET http://localhost:3001/api/entries HTTP/1.1
```

But it's possible to change the `PORT` environment variable.
In this case, the `samples.http` would not target the correct server.

You can take a look at the REST Client documentation:
https://marketplace.visualstudio.com/items?itemName=humao.rest-client

An example:
https://raw.githubusercontent.com/lmachens/aeternum-map/main/samples.http
See first three lines.

File: samples.http [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+###POST an entry
+POST http://localhost:3000/api/entries HTTP/1.1
+Content-Type: application/json
+
+{
+    "image": "content tbd",
```
Comment (lmachens) [inline samples.http:9]: I would make it more clear that you like to add a URL/src here.

```suggestion
    "image": "https://example.com/image.png",
```

```suggestion
    "imageURL": "https://example.com/image.png",
```

File: server/index.ts [hunk 3]
```diff
@@ -1,21 +1,36 @@
+import dotenv from 'dotenv';
+dotenv.config();
 import express from 'express';
-import path from 'path';
+import { connectDatabase, getEntryCollection } from './database';
+
+if (!process.env.MONGODB_URI) {
+  throw new Error('No MongoDB URI dotenv variable');
```
Comment (lmachens) [inline server/index.ts:7]:
```suggestion
  throw new Error('No MongoDB URI env variable');
```

File: server/index.ts [hunk 4]
```diff
@@ -1,21 +1,36 @@
...
 const port = process.env.PORT || 3001;
 const app = express();
+app.use(express.json());
 
 app.get('/api/hello', (_request, response) => {
   response.json({ message: 'Hello from server' });
 });
 
-// Serve production bundle
-app.use(express.static('dist'));
```
Comment (lmachens) [inline server/index.ts:12]: This was required to serve the client bundle.

**Review by lmachens — CHANGES_REQUESTED** [review]
(*^▽^*)

File: server/index.ts (hunk 4 above)
Comment (riittagerhards) [reply server/index.ts:12]: done 👍

**Pull request merged**